jrcc-access-spring-boot-autoconfigure/src/main/java/ca/bc/gov/open/jrccaccess/autoconfigure/plugins/rabbitmq/RedisStorageService.java

```diff
@@ -92,4 +92,19 @@ public String getString(String key, String digest) throws DocumentMessageExcepti
 
 	}
 
+	/**
+	 * Deletes a key/document pair from redis store
+	 * @param key
+	 * @return
+	 */
+	public Boolean deleteString(String key) throws DocumentMessageException {
+
+		try {
+                        this.cacheManager.getCache(accessProperties.getInput().getDocumentType()).evict(key);
+			return true;
+		} catch (RedisConnectionFailureException e) {
+			throw new ServiceUnavailableException("redis service unavailable", e.getCause());
```

Comment:
Yes we need an exception here. the issue with this one it will trigger the retry policy (it's bad implementation on our side I think)

it happens here:
https://github.com/bcgov/jrcc-document-access-libs/blob/f9702961b7ae70a1522e4f2922e30ec91635b2ad/jrcc-access-spring-boot-autoconfigure/src/main/java/ca/bc/gov/open/jrccaccess/autoconfigure/plugins/rabbitmq/DocumentInputErrorHandler.java#L38-L42

This means that we've successfully handled the document but then we we try to delete, the service is not up, in that case the TTL will do it's job of cleanup the document (we might want to discuss that though)

for the moment I think we should wrap the exception in the DocumentMessageException and link a new issue.

Reply:
PS there is a task to implement better exception logic